src/main/java/de/hysky/skyblocker/skyblock/dwarven/CarpetHighlighter.java

```diff
@@ -0,0 +1,109 @@
+package de.hysky.skyblocker.skyblock.dwarven;
+
+import de.hysky.skyblocker.annotations.Init;
+import de.hysky.skyblocker.config.SkyblockerConfigManager;
+import de.hysky.skyblocker.events.SkyblockEvents;
+import de.hysky.skyblocker.utils.Boxes;
+import de.hysky.skyblocker.utils.Location;
+import de.hysky.skyblocker.utils.Resettable;
+import de.hysky.skyblocker.utils.Tickable;
+import de.hysky.skyblocker.utils.render.RenderHelper;
+import de.hysky.skyblocker.utils.render.Renderable;
+import it.unimi.dsi.fastutil.objects.ObjectArraySet;
+import net.fabricmc.fabric.api.client.event.lifecycle.v1.ClientTickEvents;
+import net.fabricmc.fabric.api.client.networking.v1.ClientPlayConnectionEvents;
+import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderContext;
+import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderEvents;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.CarpetBlock;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+
+import java.awt.*;
+
+/**
+ * Highlights unbreakable carpets within ore veins in the Dwarven Mines.
+ */
+public final class CarpetHighlighter implements Renderable, Tickable, Resettable {
+	public static final CarpetHighlighter INSTANCE = new CarpetHighlighter();
+
+	private static final Vec3d CARPET_BOUNDING_BOX = Boxes.getLengthVec(CarpetBlock.SHAPE.getBoundingBox());
+	private static final int SEARCH_RADIUS = 15;
+	private static final int TICK_INTERVAL = 15;
+	private static final ObjectArraySet<BlockPos> CARPET_LOCATIONS = new ObjectArraySet<>();
+
+	private static float[] colorComponents;
+	private static int tickCounter = 0;
+	private static boolean isLocationValid = false;
+
+	@Init
+	public static void init() {
+		INSTANCE.configCallback(SkyblockerConfigManager.get().mining.dwarvenMines.carpetHighlightColor);
+		WorldRenderEvents.AFTER_TRANSLUCENT.register(INSTANCE::render);
+		SkyblockEvents.LOCATION_CHANGE.register(INSTANCE::onLocationChange);
+		ClientTickEvents.END_CLIENT_TICK.register(INSTANCE::tick);
```

Comment:
This could just be `Scheduler#scheduleCyclic`, and you would also not need the tick counter stuff.